core/src/main/java/hudson/slaves/CommandLauncher.java

```diff
@@ -167,12 +160,23 @@ public void onClosed(Channel channel, IOException cause) {
             LOGGER.log(Level.SEVERE, msg, e);
             e.printStackTrace(listener.error(msg));
 
-            if(_proc!=null)
+            if(_proc!=null) {
+                reportProcessTerminated(_proc, listener);
                 try {
                     ProcessTree.get().killAll(_proc, _cookie);
                 } catch (InterruptedException x) {
                     x.printStackTrace(listener.error(Messages.ComputerLauncher_abortedLaunch()));
                 }
+            }
+        }
+    }
+
+    private static void reportProcessTerminated(Process proc, TaskListener listener) {
+        try {
+            int exitCode = proc.exitValue();
+            listener.error("Process terminated with exit code " + exitCode);
```

Comment:
Maybe also print what process, thread id or name was terminated if such info makes sense and available?